src/tracing/swapFlowLoggers.ts

```diff
@@ -32,3 +33,25 @@ export function maybeLogFirstSwapAction(analyticsContext: ITraceContext) {
     })
   }
 }
+
+export function logSwapQuoteRequest(
+  chainId: number,
+  routerPreference: RouterPreference | typeof INTERNAL_ROUTER_PREFERENCE_PRICE
+) {
+  let performanceMetrics = {}
+  if (routerPreference !== INTERNAL_ROUTER_PREFERENCE_PRICE) {
+    const hasSetSwapQuote = tracker.hasTimestamp(SwapEventType.FIRST_QUOTE_FETCH_STARTED)
+    const elapsedTime = tracker.setElapsedTime(SwapEventType.FIRST_QUOTE_FETCH_STARTED)
+    performanceMetrics = {
+      // We only log the time_to_first_quote_request metric for the first quote request of a session.
+      time_to_first_quote_request: hasSetSwapQuote ? undefined : elapsedTime,
+      time_to_first_quote_request_since_first_input: hasSetSwapQuote
+        ? undefined
+        : tracker.getElapsedTime(SwapEventType.FIRST_QUOTE_FETCH_STARTED, SwapEventType.FIRST_SWAP_ACTION),
+    }
+  }
+  sendAnalyticsEvent(SwapEventName.SWAP_QUOTE_FETCH, {
```

Comment:
do we want to track `SWAP_QUOTE_FETCH` sent for every price fetch with just chainID and an empty obj? If not you could just have the sendAnalyticsEvent inside the `if` statement since it's a void returning fn

Reply:
yeah the idea is to send it for every quote request initiated. price requests also trigger the corresponding quote_received event so i think we should keep this.

i mostly just don't want to log time_to_first_quote for price requests because it's not really part of the UX flow that we care about